acnportal/acnsim/simulator.py

```diff
@@ -85,6 +85,8 @@ def run(self):
                     self.schedule_history[self._iteration] = new_schedule
                 self._last_schedule_update = self._iteration
                 self._resolve = False
+            self.pilot_signals = _increase_width(self.pilot_signals, self._iteration + 1)
```

Comment:
Should we use a larger step to avoid increasing the array width each iteration? Perhaps the target should once again be the timestep of the last event in the queue since this could change as unplug events are added.

Reply:
Agree. I did need to catch the case where the `event_queue` became empty (i.e. we just reached the end of the simulation) to be able to do this, so see `event_queue.py` for the changes.

Reply:
When the event queue is empty, `get_last_timestamp` returns `None`.